### PR TITLE
[AppService] functionapp: Added error message to deployment command if resource group/function name invalid

### LINKED ISSUE #11893: Azure Function - Unclear error on deployment if invalid resource group name specified.
**Describe the bug**
When executing the command below, if the specified resource group does not exist, an error is returned. The error message is very unclear as to what the problem is (invalid resource group).
`az functionapp deployment source config-zip -g "this-rg-does-not-exist" -n CSEFeedbackFunctionApp123 --src "C:\src\msft\csefeedback-function-wrong-rg-name\bin\Debug\netcoreapp2.1\MyFunction.zip"`

**To Reproduce**

- Create an Azure Function app
- Create an Azure Function (e.g. HTTP-triggered function)
- Zip the function content
- Deploy using Azure CLI

**Expected behavior**
If an invalid resource group is specified, the error message should indicate that the resource group is invalid or doesn't exist.

**Environment summary**
Windows 10 1909
Azure CLI version 2.0.80

**Additional context**
Similar to #10773 and #10639

### PR DESCRIPTION
**Pull request by gzuber**
Resolves #11893 

**History Notes:**  
[AppService] az functionapp deployment source config-zip: Add an error message if resource group or function name are invalid/don't exist.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by gzuber** [comment]
@Hazhzeng @ankitkumarr Can you take a look when you have time?

@dariagrigoriu FYI

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -4,7 +4,9 @@ Release History
 ===============
 
 **AppService**
+
 az webapp create : Fixing issue when running the command with --runtime 
+functionapp: Added error message to deployment command if resource group/function name invalid
```
Comment (yungezz) [inline src/azure-cli/HISTORY.rst]: from last release, history.rst is automatically generated, no need to manual update it. But detail change is added in PR title. Guideline here: https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests

**Comment by yungezz** [comment]
add to S166. HI @fengzhou-msft could you pls review this PR? thanks

**Comment by gzuber** [comment]
@qwordy This should be good to merge, thanks!

**Pull request merged**